pgrx-sql-entity-graph/src/pg_extern/mod.rs

```diff
@@ -230,6 +236,36 @@ impl PgExtern {
         Ok(skel)
     }
 
+    fn cast(func: &syn::ItemFn) -> syn::Result<Option<PgCast>> {
+        let mut skel = Option::<PgCast>::default();
+        for attr in &func.attrs {
+            let last_segment = attr.path.segments.last().unwrap();
+            match last_segment.ident.to_string().as_str() {
+                "pg_cast" => {
+                    let mut cast = PgCast::Default;
+                    if !attr.tokens.is_empty() {
+                        match attr.parse_args::<Expr>() {
+                            Ok(Expr::Path(p)) => {
+                                match p.path.segments.last().unwrap().ident.to_string().as_str() {
+                                    "implicit" => cast = PgCast::Implicit,
+                                    "assignment" => cast = PgCast::Assignment,
+                                    _ => eprintln!("Unrecognized option: {}. Using default cast options.", p.path.to_token_stream()),
+                                }
+                            }
+                            _ => eprintln!(
+                                "Unable to parse attribute to pg_cast as a Rust Expr: {}. Using default cast options.",
+                                attr.tokens
+                            ),
```

Comment:
I would prefer to not silently accept something that can be validated at compile time. Please return an error or panic instead.

Reply:
Done. Switched to panic on invalid options